internal/nodes/nodes.go

```diff
@@ -0,0 +1,147 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodes
+
+import (
+	"context"
+	goerrors "errors"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"strings"
+	"time"
+)
+
+// labels the specified worker node as a gateway node
+func HandleLabels(clientset kubernetes.Interface, gatewayNode struct{Node string}) error {
```

Comment:
It seems odd to pass an anonymous struct - why not just pass a string?

```suggestion
func LabelGateways(clientset kubernetes.Interface, gatewayNode struct{Node string}) error {
```